### ISSUE TITLE
Crash from Commands

### ISSUE THREAD

**Issue by Baynar** [OregonCore/OregonCore#1397]
**Revision**:
3455
**Operating system**:
Windows 10
**Current behavior**: 
....When you use .server shutdown / restart server starts shutting down or restarting
but then it crashes and i have to close it manually

**Expected behavior**: 
...Should not crash

**Steps to reproduce the problem**:
go ingame or in console type .server restart 5 or shutdown

**Comment by LordUsagi** [comment]
Are you built in debug mode or release

EDIT: And is it crashing on pressing Enter for the command? or crashing on the actual shut down? My core has been restarting just fine, with no crashes. Windows 7, Compiling 32 bit.

**Comment by Baynar** [comment]
Build in Release / Wndows 10 x64
same on Windows 10 x32 with x32 bit core release mode

Even if i press enter after i type the command in console it sits for like 10 sec and windows gives me crash error

and before that ingame the server announce is there and server goes offline after that god knows

EDIT: Without custom content fresh core is the same
EDIT: Server log http://pastebin.com/Ssxb7Edm
EDIT: seems like .server shutdown works when from console but not when ingame
tried in debug and release mode still .server restart crashes the core

**Comment by LordUsagi** [comment]
Doesn't for me. So this ones really odd. Do you actually have a crash log?

**Comment by Baynar** [comment]
There is no crash log

**Comment by kotishe** [comment]
yes crash log dont work

**Comment by Baynar** [comment]
@LordUsagi Due to the Problem i summited before a few hours i found a solution for this
disable console and restarting and shutting the server down from ingame work but u have no access to console commands sadly...

But yea they will work only if console is disabled

**Comment by kotishe** [comment]
Shutdown stop on error like ACE:mutex close clean or something like this

**Comment by LordUsagi** [comment]
I've been using server restart for days both in debug and release. No crashes at all.

**Comment by Baynar** [comment]
I type .server restart 5 ingame
i get DC
wait like 10 seconds core crash and no crash log

I type .server restart 5 ingame
i get DC
i open up the window for oregoncore.exe AND SOMETIMES like 5% it restarts but 95% of the times it crashes

I type .server restart 5 ingame
i get DC
i open up oregoncore.exe and left click in the window Server restarts normally but sometimes it decides to crash at the end or so but that is pretty rare and again no crash log

same when its from console if i don't press enter to  get out og Run Command menu core will crash

core is not crashing from the command but in the process of shutting down / trying to restart

**Comment by LordUsagi** [comment]
Not for me. You've compiled using your own MYSQL libs correct?

**Comment by Baynar** [comment]
No i use VS 2013 now, but its the same in VS 2015

**Comment by LordUsagi** [comment]
Can I have more specifics about the error message it throws? Build in debug mode and you will get a crash log.

**Comment by Desteny** [comment]
i know this as well, the problem, i noticed  it already years ago is a crash during cleanup dont writes a crashlog or dump but it happens.

if you use a auto restarter and disable windows crash window then it has no effect on your server at all, but better would be to find the reason

**Comment by Zaffy** [comment]
I tried to solve this in the past but gone to blind alley. I have really no idea why this happens. First I thought it could be obsolete ACE version but after update this still happens...

**Comment by Phentora** [comment]
I have been able to reproduce this issue with command `.server restart 5` in Debug mode ( 64-bit ). It does not generate any crash log.

I also tried `.server shutdown` and it does not crash ( at least with a crash notification ), but instead the application kinda hangs after you execute the command and if you stay in the console it doesn't shutdown unless to press something on the keyboard or on mouse click. I noticed this issue when doing `.server shutdown 1` and seeing that the application would still be running apparently after that one second has passed.

Tested on OregonCore Rev: 3869 Hash: ffc80075 (Win64,little-endian)

https://imgur.com/a/ho8bw
